chia/data_layer/data_store.py

```diff
@@ -775,12 +787,11 @@ async def delete(
         hint_keys_values: Optional[Dict[bytes, bytes]] = None,
         use_optimized: bool = True,
         *,
-        lock: bool = True,
         status: Status = Status.PENDING,
     ) -> None:
-        async with self.db_wrapper.locked_transaction(lock=lock):
+        async with self.db_wrapper.write_db():
```

Comment:
There are several examples of where you just grab the write-lock for the DB, without (seemingly) writing to it. If this is intentional and correct, I really think it warrants comments